components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/publisher/APIResourceManagerEventPublisherProxy.java

```diff
@@ -0,0 +1,304 @@
+    public void publishPostDeleteAPIScopeByScopeName(String apiResourceId, String scopeName, String tenantDomain)
+            throws APIResourceMgtException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put(IdentityEventConstants.EventProperty.API_ID, apiResourceId);
+        eventProperties.put(IdentityEventConstants.EventProperty.SCOPE_NAME, scopeName);
+        eventProperties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, tenantDomain);
+        Event event = createEvent(eventProperties, IdentityEventConstants.Event.POST_DELETE_SCOPE);
+        doPublishEvent(event);
+    }
+
+    /**
+     * Publish the pre put API resource scopes event.
+     *
+     * @param apiResourceId  API resource id.
+     * @param currentScopes  Old scopes.
+     * @param scopes         New scopes.
+     * @param tenantDomain   Tenant domain.
+     * @throws APIResourceMgtException If an error occurred while publishing the event.
+     */
+    public void publishPrePutScopes(String apiResourceId, List<Scope> currentScopes, List<Scope> scopes,
+                                    String tenantDomain) throws APIResourceMgtException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put(IdentityEventConstants.EventProperty.API_ID, apiResourceId);
+        eventProperties.put(IdentityEventConstants.EventProperty.OLD_SCOPES, currentScopes);
+        eventProperties.put(IdentityEventConstants.EventProperty.NEW_SCOPES, scopes);
+        eventProperties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, tenantDomain);
+        Event event = createEvent(eventProperties, IdentityEventConstants.Event.PRE_PUT_API_RESOURCE_SCOPES);
+        doPublishEvent(event);
+    }
+
+    /**
+     * Publish the post put API resource scopes event.
+     *
+     * @param apiResourceId  API resource id.
+     * @param currentScopes  Old scopes.
+     * @param scopes         New scopes.
+     * @param tenantDomain   Tenant domain.
+     * @throws APIResourceMgtException If an error occurred while publishing the event.
+     */
+    public void publishPostPutScopes(String apiResourceId, List<Scope> currentScopes, List<Scope> scopes,
+                                     String tenantDomain) throws APIResourceMgtException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put(IdentityEventConstants.EventProperty.API_ID, apiResourceId);
+        eventProperties.put(IdentityEventConstants.EventProperty.OLD_SCOPES, currentScopes);
+        eventProperties.put(IdentityEventConstants.EventProperty.NEW_SCOPES, scopes);
+        eventProperties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, tenantDomain);
+        Event event = createEvent(eventProperties, IdentityEventConstants.Event.POST_PUT_API_RESOURCE_SCOPES);
+        doPublishEvent(event);
+    }
+
+    private Event createEvent(Map<String, Object> eventProperties, String eventName) {
```

Comment:
Isn't it better to change the order as eventName, eventProperties

which is the same order used foe the new Event Object

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5708/commits/af6134148de86556af0c52bd598378a1852fbedd